osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -109,32 +139,54 @@ private void applyRandomisation(float rateOfChangeMultiplier, RandomObjectInfo p
 
             current.AngleRad = (float)Math.Atan2(posRelativeToPrev.Y, posRelativeToPrev.X);
 
-            var position = previous.EndPositionRandomised + posRelativeToPrev;
-
-            // Move hit objects back into the playfield if they are outside of it,
-            // which would sometimes happen during big jumps otherwise.
-            position.X = MathHelper.Clamp(position.X, 0, OsuPlayfield.BASE_SIZE.X);
-            position.Y = MathHelper.Clamp(position.Y, 0, OsuPlayfield.BASE_SIZE.Y);
-
-            current.PositionRandomised = position;
+            current.PositionRandomised = previous.EndPositionRandomised + posRelativeToPrev;
         }
 
         /// <summary>
         /// Moves the <see cref="Slider"/> and all necessary nested <see cref="OsuHitObject"/>s into the <see cref="OsuPlayfield"/> if they aren't already.
         /// </summary>
-        private void moveSliderIntoPlayfield(Slider slider, RandomObjectInfo currentObjectInfo)
+        /// <returns>The <see cref="Vector2"/> that this slider has been shifted by.</returns>
+        private Vector2 moveSliderIntoPlayfield(Slider slider, RandomObjectInfo currentObjectInfo)
         {
             var minMargin = getMinSliderMargin(slider);
 
-            slider.Position = new Vector2(
-                Math.Clamp(slider.Position.X, minMargin.Left, OsuPlayfield.BASE_SIZE.X - minMargin.Right),
-                Math.Clamp(slider.Position.Y, minMargin.Top, OsuPlayfield.BASE_SIZE.Y - minMargin.Bottom)
-            );
+            var prevPosition = slider.Position;
+
+            var newX = minMargin.Left + minMargin.Right > OsuPlayfield.BASE_SIZE.X
+                ? currentObjectInfo.PositionOriginal.X
```

Comment:
What is the special case here for? Sliders which are bigger than the playfield? I don't see this mentioned anywhere. The use of `MarginPadding` (rather than a `RectangleF`, or `Quad`, or almost *anything* else) in this context continues to be really confusing because `minMargin` is just really a bounding box but with coordinates relative to the slider origin point.

Reply:
The special case is for sliders bigger than the playfield, since `Math.Clamp` will throw an exception in this case. This special case should really be handled in `getMinSliderMargin` instead. Will refactor along with changing the `MarginPadding`.